src/Compilers/Core/MSBuildTask/Microsoft.Build.Tasks.CodeAnalysis.csproj

```diff
@@ -100,11 +100,11 @@
     <PropertyGroup>
       <CompilerApiVersion>$([System.Version]::Parse($(VersionPrefix)).Major).$([System.Version]::Parse($(VersionPrefix)).Minor)</CompilerApiVersion>
       <CompilerVersionTargetsFileContent>
-<![CDATA[<Project>
+<Project>
   <PropertyGroup>
     <CompilerApiVersion>roslyn$(CompilerApiVersion)</CompilerApiVersion>
   </PropertyGroup>
-</Project>]]>
+</Project>
```

Comment:
Doesn't this change though make this a part of the msbuild project file? The intent here was for this to be data in the property that was never interpreted by MSBuild. Now it's seemingly a part of the normal XML hence valid for MSBuild interpretation.

Reply:
msbuild supports arbitrary XML as a value of the property. Even with CDATA the value is interpreted -- i.e. `$(CompilerApiVersion)` is substituted with its value.

Reply:
I'm still confused why this is a bug. It's valid XML correct? If so then how is it a bug to have this in a project file? 

@rainersigwald

Reply:
I am also confused; it looks like we do The Right Thing in a reduced case:

```xml
<Project>
  <PropertyGroup>
        <Prop>
<![CDATA[<Project>
  <PropertyGroup>
    <CompilerApiVersion>roslyn$(CompilerApiVersion)</CompilerApiVersion>
  </PropertyGroup>
</Project>]]>
      </Prop>
    </PropertyGroup>

<Target Name="PrintStuff">
  <Message Text="Prop: $(Prop)" />

  <PropertyGroup>
        <TargetProp>
<![CDATA[<Project>
  <PropertyGroup>
    <CompilerApiVersion>roslyn$(CompilerApiVersion)</CompilerApiVersion>
  </PropertyGroup>
</Project>]]>
      </TargetProp>
    </PropertyGroup>

  <Message Text="TargetProp: $(TargetProp)" />
</Target>
</Project>
```

```sh-session
❯ msbuild .\test.proj
MSBuild version 17.4.0-preview-22477-01+fcc478938 for .NET Framework
Build started 9/30/2022 2:07:51 PM.
Project "S:\repro\dotnet\roslyn\pull\64334\test.proj" on node 1 (default targets).
PrintStuff:
  Prop: <Project>
    <PropertyGroup>
      <CompilerApiVersion>roslyn</CompilerApiVersion>
    </PropertyGroup>
  </Project>
  TargetProp: <Project>
    <PropertyGroup>
      <CompilerApiVersion>roslyn</CompilerApiVersion>
    </PropertyGroup>
  </Project>
Done Building Project "S:\repro\dotnet\roslyn\pull\64334\test.proj" (default targets).


Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:00.05
```

Reply:
It also appears to work in a build of this project (without this change):

```
WriteLinesToFile
    Parameters
        Lines = <Project>
  <PropertyGroup>
    <CompilerApiVersion>roslyn4.5</CompilerApiVersion>
  </PropertyGroup>
</Project>
        Overwrite = True
        File = S:\roslyn\artifacts\obj\Microsoft.Build.Tasks.CodeAnalysis\Debug\Microsoft.Managed.Core.CurrentVersions.targets
```

Do you have a binlog of when it didn't work @tmat?

Reply:
I do not currently. I can try to get one.